### PR TITLE
3 tests

### PR DESCRIPTION
**Pull request by szepeviktor**
1. syntax check
1. coding standard
1. static analysis

Was https://bitbucket.org/towa_gmbh/towa-gdpr-plugin/pull-requests/15

### CODE REVIEW

**Comment by szepeviktor** [comment]
[bitbucket-pipelines.yml](https://github.com/towa-digital/towa-gdpr-plugin/pull/5/files#diff-81f62e874f3f2d77c10287105a377dce) should be converted to GitHub Actions ❗️ 

Okay?

**Comment by Martin-Welte** [comment]
@szepeviktor thanks for adding the PR again. Next step will be to add github pipelines :)

**Comment by szepeviktor** [comment]
> to add github pipelines :)

No-no :)

GitHub Actions!

**Comment by szepeviktor** [comment]
You may be Atlassian-infected :)

**Review by SahinU88 — CHANGES_REQUESTED** [review]
Can we resolve the conflicts so we can merge this?

**Comment by szepeviktor** [comment]
Yes!

**Comment by szepeviktor** [comment]
Done 🍏

**Review by SahinU88 — CHANGES_REQUESTED** [review]
Just saw, that the vendor-folder is not excluded by the tests/linting, that should be included. We don't have to include external resources.

**Comment by szepeviktor** [comment]
Please help me.

In php-lint there is `-path ./vendor -prune` which excludes the vendor-folder.

**Comment by szepeviktor** [comment]
@SahinU88 Please help me understand.

**Comment by SahinU88** [comment]
@szepeviktor 
sry for my late respone, kind missed your comment.
If you run `composer test` you can see that the vendor folder is included in the linter-tests etc.

image

We should exclude the vendor folder.

**Comment by szepeviktor** [comment]
This is **impossible**
Working on it right now...

**Comment by SahinU88** [comment]
@szepeviktor 
also small question, do we needd the xml-linter in this repo? we ddont have any xml-files actually expect the  phpunit-test xml file. guess we could leave it out.

and also, just comes to my mind, why do we mix php-lint and php-cs-fixer? we normally go with php-cs-fixer, then I guess the php-lint would be not needed anymore.

let me know what the reasson was, otherweise we can conssider removing those two dependencies.

**Comment by szepeviktor** [comment]
Every thing is fixed.

It should output

```
viktor@mail:~/src/towa-gdpr-plugin$ composer test
> ./bin/php-lint
No syntax errors detected in ./tests/Unit/PluginTest.php
No syntax errors detected in ./tests/bootstrap.php
No syntax errors detected in ./tests/TestCase.php
No syntax errors detected in ./config/defaults.php
No syntax errors detected in ./uninstall.php
No syntax errors detected in ./towa-gdpr-plugin.php
No syntax errors detected in ./init.php
No syntax errors detected in ./src/Consentlogger/Consent.php
No syntax errors detected in ./src/Acf/AcfGroupInterface.php
No syntax errors detected in ./src/Acf/AcfCookies.php
No syntax errors detected in ./src/Acf/AcfSettings.php
No syntax errors detected in ./src/Acf/AcfUtility.php
No syntax errors detected in ./src/SettingsTableAdapter.php
No syntax errors detected in ./src/Plugin.php
No syntax errors detected in ./src/Helper/PluginHelper.php
No syntax errors detected in ./src/Hash.php
No syntax errors detected in ./src/Rest/Rest.php
```

**Comment by szepeviktor** [comment]
> do we needd the xml-linter in this repo? we ddont have any xml-files actually expect the phpunit-test xml file. guess we could leave it out.

My QA strategy is "Check every single file." ©️ 
Now `./bin/xml-lint` checks
- phpcs.xml.dist
- phpunit.xml.dist

and both of them from the _root_ XMLSchema.xsd

> why do we mix php-lint and php-cs-fixer? we normally go with php-cs-fixer, then I guess the php-lint would be not needed anymore.

as in the [top comment](https://github.com/towa-digital/towa-gdpr-plugin/pull/5#issue-388495571) there are 3 basic tests: syntax, coding style and static analysis, `bin/php-lint` looks for syntax errors, phpcs for coding style, phpstan does static analysis

Would you like me to add more details?

**Review by SahinU88 — CHANGES_REQUESTED** [review]
The `post-install-cmd` and `post-update-cmd` are not working:

image

and just as an idea, would it be useful just using the insights package from nuno-maduro?

otherwise, the config is fine for the time-being, I would argue that we dont need the xml-linters. they will change rarely and I don't see the benefit from the linter at the moment.

**Comment by szepeviktor** [comment]
> would it be useful just using the insights package from nuno-maduro?

php-insights is a kitten, phpstan is a full-blown Tiger - You choose!

> I would argue that we dont need the xml-linters

My policy is to check each-and-everything as it is beneficial in the long run. Please decide.

> The post-install-cmd and post-update-cmd are not working

Are you using a UNIX-y OS?

**Comment by szepeviktor** [comment]
I see!! A Mac! https://stackoverflow.com/a/21243084

Let me fix it for you.

**Comment by SahinU88** [comment]
Thx, the script works now.

But seems like the vendor folders are still included. Do I have to udpate something or delete any cache-config files or such?

**Comment by szepeviktor** [comment]
messages starting with "No syntax error" come from this command.

```shell
# Find all files
#     exclude vendor, bin, .git
#     only .php files
# Run syntax check for each file
find ./ \
    -path ./vendor -prune -o -path ./bin -prune -o -path ./.git -prune -o \
    -name "*.php" -print0 \
    | xargs -0 -L 1 -P 4 -- php -l -f
```

I suspect another Mac issue!
Could you resolve it as I don't have access to OSX shell?

**Comment by szepeviktor** [comment]
```shell
find ./ -type f -name "*.php" -o -path ./vendor -prune -o -path ./bin -prune -o -path ./.git -prune
```

Does this command **exclude** the vendor dir?

**Comment by szepeviktor** [comment]
Got it from your screenshot :) https://github.com/towa-digital/towa-gdpr-plugin/pull/5#issuecomment-610570766
Fixing double slashes ...

**Comment by szepeviktor** [comment]
All done 🍏

File: src/Helper/PluginHelper.php [hunk 1]
```diff
@@ -25,7 +25,7 @@ public static function isGdprPluginAdminScreen()
     {
         $screen = \get_current_screen();
 
-        return ($screen && Str::contains($screen, 'towa-gdpr-plugin'));
+        return ($screen instanceof \WP_Screen && Str::contains($screen->id, 'towa-gdpr-plugin'));
```
Comment (szepeviktor) [inline src/Helper/PluginHelper.php:28]: @SahinU88 @phpstan discovered a critical error here

Comment (Martin-Welte) [reply src/Helper/PluginHelper.php:28]: hahah i noticed it as well. But this check is way better :)

Comment (Martin-Welte) [reply src/Helper/PluginHelper.php:28]: @volkanakb can you make a hotfix for this?

Comment (szepeviktor) [reply src/Helper/PluginHelper.php:28]: @volkanakb Your git email is not registered here on GitHub.

**Comment by szepeviktor** [comment]
Could we get this one merged?

**Comment by SahinU88** [comment]
We will stop the maintenance of this in the near future, thus we don't look at requests.

Sry for this one.

**Comment by szepeviktor** [comment]
@SahinU88 @Martin-Welte What happened to the project? Isn't there a need for a general GDPR plugin?

**Comment by SahinU88** [comment]
@szepeviktor we decided not to maintain this package anymore, because there are a lot of solutions for WordPress anyway which are good and are developed continously.

Thanks for your support and your contributions. We will make this repo read-only.

Thx

**Pull request closed without merging**